File: src/main/java/pl/training/workshop/m6/s01_strategy/step3/PremiereDiscount.java
package pl.training.workshop.m6.s01_strategy.step3;

import pl.training.workshop.shared.Money;

/** Krok 3: premiera - brak zniżek (typ biletu nie jest nawet sprawdzany, jak w start). */
public final class PremiereDiscount implements DiscountPolicy {
    @Override
    public Money discount(Money base, String ticketType) {
        return Money.ZERO;
    }
}

File: csharp/src/Training.Workshop/M6/S01Strategy/Step3/PremiereDiscount.cs
using Training.Workshop.Shared;

namespace Training.Workshop.M6.S01Strategy.Step3;

/// <summary>Krok 3: premiera - brak zniżek (typ biletu nie jest nawet sprawdzany, jak w Start).</summary>
public sealed class PremiereDiscount : IDiscountPolicy
{
    public Money Discount(Money basePrice, string ticketType)
    {
        return Money.Zero;
    }
}
